contracts/examples/ProvideDataExamples/interestRateOracleContract.sol

```diff
@@ -0,0 +1,233 @@
+    function getSupplyBalance(address account, address asset) external view returns (uint);
+    function borrowRatePerBlock() external view returns (uint);
+    function supplyRatePerBlock() external view returns (uint);
+    function mint(uint mintAmount) external returns (uint);
+    function redeem(uint redeemTokens) external returns (uint);
+    function balanceOf(address account) external view returns (uint);
+}
+
+// Fulcrum
+interface Fulcrum {
+    function borrowInterestRate() external view returns (uint256);
+    function supplyInterestRate() external view returns (uint256);
+}
+
+interface DyDx {
+    struct val {
+        uint256 value;
+    }
+
+    struct set {
+        uint128 borrow;
+        uint128 supply;
+    }
+
+    function getEarningsRate() external view returns (val memory);
+    function getMarketInterestRate(uint256 marketId) external view returns (val memory);
+    function getMarketTotalPar(uint256 marketId) external view returns (set memory);
+}
+
+interface LendingPoolAddressesProvider {
+    function getLendingPoolCore() external view returns (address);
+}
+
+interface LendingPoolCore  {
+    function getReserveCurrentLiquidityRate(address _reserve)
+    external
+    view
+    returns (
+        uint256 liquidityRate
+    );
+    function getReserveCurrentVariableBorrowRate(address _reserve) external view returns (uint256);
+}
+
+interface OrFeedInterface {
+    function getExchangeRate (string fromSymbol, string toSymbol, string venue, uint256 amount) external view returns (uint256);
+    function getTokenDecimalCount (address tokenAddress) external view returns (uint256);
+    function getTokenAddress (string symbol) external view returns (address);
+    function getSynthBytes32 (string symbol) external view returns (bytes32);
+    function getForexAddress (string symbol) external view returns (address);
+}
+
+contract InterestRateOracleContract {
+    struct PlatformSide {
+        string platform;
+        bool isBorrow;
+    }
+
+    OrFeed orfeed = OrFeedInterface(0x8316b082621cfedab95bf4a44a1d4b64a6ffc336);
+    DyDx dYdX = DyDx(0x1E0447b19BB6EcFdAe1e4AE1694b0C3659614e4e);
+    LendingPoolCore aave = LendingPoolCore(LendingPoolAddressesProvider(0x24a42fD28C976A61Df5D00D0599C34c4f90748c8).getLendingPoolCore());
+
+    uint256 constant ASCII_HYPHEN = 45;
+    uint256 constant ASCII_ZERO = 48;
+    uint256 constant ASCII_UPPERCASE_X = 88;
+    uint256 constant ASCII_LOWERCASE_X = 120;
+    uint256 constant DECIMAL = 10 ** 18;
+
+    constructor() public payable {
+        owner = msg.sender;
+    }
+
+    function getInterestRate(string fromParam, string, string venue, uint256 amount) public returns (uint256) {
+        string memory tokenSymbol = getTokenSymbol(fromParam);
+        PlatformSide memory platformSide = getPlatformSide(venue);
+        uint256 memory interestRate = getInterestRate(tokenSymbol, platformSide);
+
+        return amount.mul(interestRate).div(DECIMAL);

```

Comment:
I'm not 100% sure that this is the correct method of calculation, but I did some unit analysis and made an educated guess.